storage/range_tree.go

```diff
@@ -20,29 +20,286 @@ package storage
 import (
 	"github.com/cockroachdb/cockroach/proto"
 	"github.com/cockroachdb/cockroach/storage/engine"
+	"github.com/cockroachdb/cockroach/util"
 )
 
-// SetupRangeTree creates a new range tree.  This should only be called as part of
-// store.BootstrapRange.
-func SetupRangeTree(batch engine.Engine, ms *engine.MVCCStats,
-	timestamp proto.Timestamp, startKey proto.Key) error {
+type (
+	// RangeTree represents holds the metadata about the RangeTree.  There is
+	// only one for the whole cluster.
+	RangeTree proto.RangeTree
```

Comment:
I would actually suggest making this a structure with an embedded proto.RangeTree, and including the engine, mvccstats and timestamp along with it.  Like:
```
type RangeTreeNode struct {
  proto.RangeTreeNode
  engine         engine.Engine
  ms               *engine.MVCCStats
  timestamp   proto.Timestamp
}
```
These are getting passed to all of the methods, and I'm assuming they don't change between calls to the methods.  Encapsulating these should make these signatures much smaller, and should make the code much cleaner when the struct calls its own methods.